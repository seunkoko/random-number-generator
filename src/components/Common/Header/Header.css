.navbar {
  border-radius: 0;
  color: #F2F2F2 !important;
}

.container-fluid {
  width: 100%;
}

.header__container {
  font-size: 0.8rem;
}

.nav {
  float: right;
  letter-spacing: 0.1rem;
}

.navbar-brand {
  color: #F2F2F2 !important;
  font-size: 1rem;
  letter-spacing: 0.1rem;
}

.__logo {
  float: left;
  margin: 0.5rem 1rem 0 0;
  width: 3rem;
  height: 2rem;
}

.navbar-nav>.active>a {
  color: #F2F2F2 !important;
}

.navbar-nav>.active>a:focus {
  color: #F2F2F2 !important;
}

.navbar-nav>.active>a:hover {
  color: #F2F2F2 !important;
}

