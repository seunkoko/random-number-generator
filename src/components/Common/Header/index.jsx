import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import './Header.css';

// logo
import logo from '../../../assets/images/rng-logo-black.png';

// utils
import baseRouterProps from '../../../utils/baseRouterProps';

/**
 * Header - Header component
 *
 * @class Header
 * @param {object} props - props
 */
const Header = (props) => {
  const {
    homeActive,
    operationsActive,
  } = props;

  return (
    <nav className="navbar navbar-inverse">
      <div className="container-fluid">
        <div className="container header__container">
          <div className="navbar-header">
            <img alt="" className="__logo" src={logo} />
            <a className="navbar-brand" href="/">Random Number Generator</a>
          </div>
          <ul className="nav navbar-nav">
            <li
              className={homeActive ? 'active' : ''}
            >
              <a id="homeTab" href="/">Home</a>
            </li>
            <li
              className={operationsActive ? 'active' : ''}
            >
              <a id="operationsTab" href="/operations">Operations</a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

Header.propTypes = {
  ...baseRouterProps,
  homeActive: PropTypes.bool.isRequired,
  operationsActive: PropTypes.bool.isRequired,
};

export default Header;
