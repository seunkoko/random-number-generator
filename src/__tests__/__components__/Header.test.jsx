// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import Header from '../../components/Common/Header';

describe('Header', () => {
  const props = {
    homeActive: false,
    operationsActive: true,
  };

  it('renders properly', () => {
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper.find('nav').length).toEqual(1);
    expect(wrapper.find('ul').length).toEqual(1);
    expect(wrapper.find('li').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });

  it('should reload to home page when the Home Tab is clicked', () => {
    const wrapper = shallow(<Header {...props} />);
    wrapper.find('#homeTab').simulate('click');
  });

  it('should redirect to the operations page when the Operations Tab is clicked', () => {
    const wrapper = shallow(<Header {...props} />);
    wrapper.find('#operationsTab').simulate('click');
  });
});
