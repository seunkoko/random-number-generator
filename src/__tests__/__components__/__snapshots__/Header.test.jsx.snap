// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    homeActive={false}
    operationsActive={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid"
>
        <div
                className="container header__container"
        >
                <div
                        className="navbar-header"
                >
                        <img
                                alt=""
                                className="__logo"
                                src="rng-logo-black.png"
                        />
                        <a
                                className="navbar-brand"
                                href="/"
                        >
                                Random Number Generator
                        </a>
                </div>
                <ul
                        className="nav navbar-nav"
                >
                        <li
                                className=""
                        >
                                <a
                                        href="/"
                                        id="homeTab"
                                >
                                        Home
                                </a>
                        </li>
                        <li
                                className="active"
                        >
                                <a
                                        href="/operations"
                                        id="operationsTab"
                                >
                                        Operations
                                </a>
                        </li>
                </ul>
        </div>
</div>,
      "className": "navbar navbar-inverse",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container header__container"
>
          <div
                    className="navbar-header"
          >
                    <img
                              alt=""
                              className="__logo"
                              src="rng-logo-black.png"
                    />
                    <a
                              className="navbar-brand"
                              href="/"
                    >
                              Random Number Generator
                    </a>
          </div>
          <ul
                    className="nav navbar-nav"
          >
                    <li
                              className=""
                    >
                              <a
                                        href="/"
                                        id="homeTab"
                              >
                                        Home
                              </a>
                    </li>
                    <li
                              className="active"
                    >
                              <a
                                        href="/operations"
                                        id="operationsTab"
                              >
                                        Operations
                              </a>
                    </li>
          </ul>
</div>,
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="navbar-header"
>
              <img
                            alt=""
                            className="__logo"
                            src="rng-logo-black.png"
              />
              <a
                            className="navbar-brand"
                            href="/"
              >
                            Random Number Generator
              </a>
</div>,
            <ul
              className="nav navbar-nav"
>
              <li
                            className=""
              >
                            <a
                                          href="/"
                                          id="homeTab"
                            >
                                          Home
                            </a>
              </li>
              <li
                            className="active"
              >
                            <a
                                          href="/operations"
                                          id="operationsTab"
                            >
                                          Operations
                            </a>
              </li>
</ul>,
          ],
          "className": "container header__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="__logo"
                  src="rng-logo-black.png"
/>,
                <a
                  className="navbar-brand"
                  href="/"
>
                  Random Number Generator
</a>,
              ],
              "className": "navbar-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "__logo",
                  "src": "rng-logo-black.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Random Number Generator",
                  "className": "navbar-brand",
                  "href": "/",
                },
                "ref": null,
                "rendered": "Random Number Generator",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className=""
>
                  <a
                                    href="/"
                                    id="homeTab"
                  >
                                    Home
                  </a>
</li>,
                <li
                  className="active"
>
                  <a
                                    href="/operations"
                                    id="operationsTab"
                  >
                                    Operations
                  </a>
</li>,
              ],
              "className": "nav navbar-nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="/"
                    id="homeTab"
>
                    Home
</a>,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "href": "/",
                    "id": "homeTab",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="/operations"
                    id="operationsTab"
>
                    Operations
</a>,
                  "className": "active",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Operations",
                    "href": "/operations",
                    "id": "operationsTab",
                  },
                  "ref": null,
                  "rendered": "Operations",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid"
>
          <div
                    className="container header__container"
          >
                    <div
                              className="navbar-header"
                    >
                              <img
                                        alt=""
                                        className="__logo"
                                        src="rng-logo-black.png"
                              />
                              <a
                                        className="navbar-brand"
                                        href="/"
                              >
                                        Random Number Generator
                              </a>
                    </div>
                    <ul
                              className="nav navbar-nav"
                    >
                              <li
                                        className=""
                              >
                                        <a
                                                  href="/"
                                                  id="homeTab"
                                        >
                                                  Home
                                        </a>
                              </li>
                              <li
                                        className="active"
                              >
                                        <a
                                                  href="/operations"
                                                  id="operationsTab"
                                        >
                                                  Operations
                                        </a>
                              </li>
                    </ul>
          </div>
</div>,
        "className": "navbar navbar-inverse",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container header__container"
>
            <div
                        className="navbar-header"
            >
                        <img
                                    alt=""
                                    className="__logo"
                                    src="rng-logo-black.png"
                        />
                        <a
                                    className="navbar-brand"
                                    href="/"
                        >
                                    Random Number Generator
                        </a>
            </div>
            <ul
                        className="nav navbar-nav"
            >
                        <li
                                    className=""
                        >
                                    <a
                                                href="/"
                                                id="homeTab"
                                    >
                                                Home
                                    </a>
                        </li>
                        <li
                                    className="active"
                        >
                                    <a
                                                href="/operations"
                                                id="operationsTab"
                                    >
                                                Operations
                                    </a>
                        </li>
            </ul>
</div>,
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="navbar-header"
>
                <img
                                alt=""
                                className="__logo"
                                src="rng-logo-black.png"
                />
                <a
                                className="navbar-brand"
                                href="/"
                >
                                Random Number Generator
                </a>
</div>,
              <ul
                className="nav navbar-nav"
>
                <li
                                className=""
                >
                                <a
                                                href="/"
                                                id="homeTab"
                                >
                                                Home
                                </a>
                </li>
                <li
                                className="active"
                >
                                <a
                                                href="/operations"
                                                id="operationsTab"
                                >
                                                Operations
                                </a>
                </li>
</ul>,
            ],
            "className": "container header__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="__logo"
                    src="rng-logo-black.png"
/>,
                  <a
                    className="navbar-brand"
                    href="/"
>
                    Random Number Generator
</a>,
                ],
                "className": "navbar-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "__logo",
                    "src": "rng-logo-black.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Random Number Generator",
                    "className": "navbar-brand",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": "Random Number Generator",
                  "type": "a",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className=""
>
                    <a
                                        href="/"
                                        id="homeTab"
                    >
                                        Home
                    </a>
</li>,
                  <li
                    className="active"
>
                    <a
                                        href="/operations"
                                        id="operationsTab"
                    >
                                        Operations
                    </a>
</li>,
                ],
                "className": "nav navbar-nav",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="/"
                      id="homeTab"
>
                      Home
</a>,
                    "className": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Home",
                      "href": "/",
                      "id": "homeTab",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": "a",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="/operations"
                      id="operationsTab"
>
                      Operations
</a>,
                    "className": "active",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Operations",
                      "href": "/operations",
                      "id": "operationsTab",
                    },
                    "ref": null,
                    "rendered": "Operations",
                    "type": "a",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
